### PR TITLE
:book: Fix Equinix metal link in docs

### PR DESCRIPTION
**Pull request by killianmuldoon**
Fix a broken link in the docs.

/area documentation

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -539,7 +539,7 @@ clusterctl init --infrastructure docker
 In order to initialize the Equinix Metal Provider (formerly Packet) you have to expose the environment
 variable `PACKET_API_KEY`. This variable is used to authorize the infrastructure
 provider manager against the Equinix Metal API. You can retrieve your token directly
-from the [Equinix Metal Console](https://console.equinix.com/).
```
Comment (killianmuldoon) [inline docs/book/src/user/quick-start.md:542]: @displague I think you added this in https://github.com/kubernetes-sigs/cluster-api/pull/5774. Do you know if there's any alternative link? This one is now giving a 403 for the MD link checker. I'd prefer to remove it if there's no easy substitution.

**Comment by fabriziopandini** [comment]
If this is creating noise on PRs, let's merge as it is, we can always add the link later

**Comment by killianmuldoon** [comment]
> If this is creating noise on PRs, let's merge as it is, we can always add the link later

Generally agreed - currently we don't run the full link checker on PRs so this is just failing the weekly job. I'm fine to merge as is though.

**Comment by killianmuldoon** [comment]
@fabriziopandini @chrischdi mind putting lgtm on this one?

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Comment by killianmuldoon** [comment]
/cherry-pick release-1.4

These cherry-picks failed with the issues [k8s-infra-cherrypick-robot](https://github.com/k8s-infra-cherrypick-robot) had. Should be fixed now.

**Pull request merged**